src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketGitSCMRevision.java

```diff
@@ -96,18 +96,18 @@ public boolean equals(Object o) {
         if (this == o) {
             return true;
         }
-        if (o == null || !(o instanceof SCMRevisionImpl)) {
+        if (o == null || getClass() != o.getClass()) {
```

Comment:
here is needed to be `o instanceof SCMRevisionImpl` because commit stored in the job could be a GitCommitRevision (I do not remember the exact the class) or trait/build strategy will fail. Please revert

Reply:
I'll try to have a look at it in the next days.

Reply:
It's not so easy provide a test case. I got it at the time of develop of commit message/author. The class must be backward compatible with the previous implementation (SCMRevisionImpl) or who upgrade the plugin will have strange behavior.
For the test case we need an existing workspace with a job where at least a build was run (with a plugin version previous the creation of this class) that had stored the SHA of the last built commit (in the scm-revision-hash.xml).

Add a comment to like `// for backward compatiblity we need to compare against a serialisation of the previous implementation of bitbucket revision`